FlowerOrders/FlowerOrders/Screens/OrdersList/OrderListViewController.swift

```diff
@@ -0,0 +1,202 @@
+//
+//  OrderListViewController.swift
+//  FlowerOrders
+//
+//  Created by Paul Tiriteu on 05/08/2020.
+//  Copyright © 2020 Paul Tiriteu. All rights reserved.
+//
+
+import UIKit
+
+
+
+class OrderListViewController: UIViewController {
+    private let viewModel: OrderListViewModel
+    
+    private lazy var searchBar: UISearchBar = {
+        let searchBar = UISearchBar()
+        searchBar.searchBarStyle = .minimal
+        searchBar.placeholder = "Search..."
+        searchBar.sizeToFit()
+        searchBar.backgroundColor = .white
+        searchBar.returnKeyType = .done
+        searchBar.enablesReturnKeyAutomatically = false
+        searchBar.delegate = self
+        
+        return searchBar
+    }()
+    
+    private lazy var tableView: UITableView = {
+        let tableView = UITableView()
+        tableView.register(UINib(nibName: OrderTableViewCell.className, bundle: Bundle(for: OrderTableViewCell.self)), forCellReuseIdentifier: OrderTableViewCell.className)
+        tableView.separatorStyle = .none
+        
+        return tableView
+    }()
+    
+    private lazy var refreshControl: UIRefreshControl = {
+        let refreshControl = UIRefreshControl()
+        refreshControl.addTarget(self, action: #selector(refreshTableView), for: .valueChanged)
+        refreshControl.tintColor = .black
+        return refreshControl
+    }()
+    
+    init(viewModel: OrderListViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        title = "Your orders list"
+        navigationController?.navigationBar.standardAppearance.backgroundColor = .white
+        navigationController?.navigationBar.standardAppearance.shadowColor = .clear
+        
+        configureSearchBar()
+        configureTableView()
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(displayLocalOrders), name: Notification.Name.NSManagedObjectContextDidSave, object: nil)
```

Comment:
Interesting approach 👍 any particular reason for using this instead of the delegate? I saw you're using the delegate to update the list in other cases. I know that it would require you to pass the list from the Repository to the ViewModel, since you don't do that now, but are there any other reasons for this approach?

Reply:
I added this notification because it's called whenever we save the context and I believe this is a scalable approach, and we won't have to pass the response every time we do a request. This would be for future implementations

Reply:
Yes, it makes total sense, thanks for clarifying.